src/main/haskell/kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -599,6 +599,45 @@ test_andTermsSimplification = give mockSymbolOrAliasSorts
                     (Mock.builtinMap [(Mock.aConcrete, fOfA), (Mock.bConcrete, fOfB)])
                 )
         )
+    , testCase "builtin List domain"
```

Comment:
Could you add test cases for:
* Unification returning a non-empty substitution
* Unification that returns an error (unifying `Mock.plain10(Mock.x)` with `Mock.a` should fail)
* Unification that merges two substitutions for the same variable, discovering a new substitution for another variable (`Mock.x=Mock.functionalconstr10(Mock.a)` with `Mock.x=Mock.functionalconstr10(Mock.y)`)

Reply:
On it

Reply:
Well, I'm not sure if those fit the definition of a unit test, which is the smallest possible test of the new functionality. What I'm trying to do right now is add a unit test for list builtins that use a hooked "concat" symbol.

Reply:
Ok, fine. Then, the following seem part of the list unification functionality, please unit-test all of them as you see fit.
* List unification can return a non-empty substitution.
* List unification returns an error because child unification fails.
* List unification merges child substitutions.
* List unification normalizes the merged child substitution.

Reply:
Oh, you're saying basically to test that all possible results of unification "pass through" the list unification. This is precisely why we try to use combinators:

- List unification returns an error because child unification fails.

Should be automatic from use of the Result monad.

- List unification merges child substitutions.
- List unification normalizes the merged child substitution.

Should be automatic from the use of ExpandedPattern's `Applicative` instance, though that's not 100% fixed yet.

Reply:
Well, there already is one test that tries to unify `[a, a]` with `[a, b]` and expects an error. Technically that isn't even the smallest "unit" to test, `[a] /\ [b]` is...

Reply:
I don't think list unification should propagate the `\bottom` from a failed child to the top. The simplifier already does that, and I don't think we should implement the same functionality twice. (I'm adding tests for it in #274.)